astropy/modeling/tests/test_compound.py

```diff
@@ -919,3 +920,149 @@ def test_compound_evaluate_fix_inputs_by_position():
         compound.evaluate(x, y, *model_params),
         model.evaluate(x + 5, y, *model_params),
     )
+
+
+@pytest.mark.skipif('not HAS_SCIPY')
+def test_fit_multiplied_compound_model_with_mixed_units():
+    """
+    Regression test for issue #12320
+    """
+
+    fitter = LevMarLSQFitter()
+    x = np.linspace(0, 1, 101) * u.s
+    y = np.linspace(5, 10, 101) * u.m * u.kg / u.s
+
+    m1 = Linear1D(slope=5*u.m/u.s/u.s, intercept=1.0*u.m/u.s)
+    m2 = Linear1D(slope=0.0*u.kg/u.s, intercept=10.0*u.kg)
+    truth = m1 * m2
+    fit = fitter(truth, x, y)
+
+    unfit_output = truth(x)
+    fit_output = fit(x)
+
+    assert unfit_output.unit == fit_output.unit == (u.kg * u.m / u.s)
+    assert_allclose(unfit_output, fit_output)
+
+    for name in truth.param_names:
+        assert getattr(truth, name) == getattr(fit, name)
```

Comment:
This is asserting that the units are properly handled, but not that an actual fit is returned. If `fit` was the model fitted to `y`, `fit(x)` should return values close to `y`. Instead it returns results identical to the init model, and in fact the parameter check shows that the function parameters are unchanged from their starting values.